lib/archethic/transaction_chain/transaction_summary.ex

```diff
@@ -252,7 +252,7 @@ defmodule Archethic.TransactionChain.TransactionSummary do
         end
       end,
       on_timeout: :kill_task,
-      max_concurrency: length(addresses)
+      max_concurrency: max(System.schedulers_online(), length(addresses))
```

Comment:
There is a pattern matching in function definition ensuring addresses is not empty